### DISCUSSION TITLE
Opentelemetry, aggregate updates

### DISCUSSION THREAD

**Discussion by eugene-bk-eng** [open-telemetry/opentelemetry-java#4321] [Q&A]
Hi, I would like to aggregate telemetry updates for method foo() into statistical interval (min/max/avg/first/last/count) and send say every minute instead of on each method call. I'd like avoid telemetry overheard on critical methods. Is there way to annotate the method to do so or would you suggest a custom tracer? If so, how. thanks

**Comment by trask** [comment]
hi @eugene-bk-eng! it probably makes sense to use the Metrics API for this.

(there's some initial work on adding annotation to generate metrics in #4260, if you are using the opentelemetry javaagent)

**Reply by eugene-bk-eng** [reply]
Is there any example to get me started with Metrics API to implement my use case? thx
